### PR TITLE
cmdb api search update

### PR DESCRIPTION
**Pull request by nameisaravind**
cmdb api search by ci/cikey/legacyName/commonName

### CODE REVIEW

**Review by nireeshT — APPROVED** [review]
LGTM

File: src/main/java/com/capitalone/dashboard/service/TestResultServiceImpl.java [hunk 1]
```diff
@@ -604,7 +604,8 @@ private String getConfigurationItem(String configurationItem, String targetAppNa
             return configurationItem;
         }
         else if (StringUtils.isNotBlank(targetAppName) ){
-           List<Cmdb> cmdb = cmdbService.commonNameByConfigurationItem(targetAppName);
+            List<Cmdb> cmdb = cmdbService.configurationItemsByTypeWithFilter("", targetAppName,
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/service/TestResultServiceImpl.java:607]: Why is the page size set to this large value? it needs to be something manageable

File: src/main/java/com/capitalone/dashboard/service/CmdbServiceImpl.java [hunk 2]
```diff
@@ -27,10 +27,10 @@ public CmdbServiceImpl(CmdbRepository cmdbRepository) {
     public Page<Cmdb> configurationItemsByTypeWithFilter(String itemType, String filter, Pageable pageable) {
 
         Page<Cmdb> configItemString;
-
         if( StringUtils.isNotEmpty( filter ) ){
 
-            List<Cmdb> cmdbList = cmdbRepository.findAllByConfigurationItemContainingOrCommonNameContainingAllIgnoreCase(filter,filter);
+            List<Cmdb> cmdbList = cmdbRepository.findAllByConfigurationItemContainingOrConfigurationKeyContainingOrLegacyServiceManagerNameContainingOrCommonNameContainingAllIgnoreCase
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/service/CmdbServiceImpl.java:32]: can you check the performance of this call? I doubt it might make it slow

**Review by rvema — CHANGES_REQUESTED** [review]
please address the review comments

File: src/main/java/com/capitalone/dashboard/service/CmdbServiceImpl.java (hunk 2 above)
Comment (nameisaravind) [reply src/main/java/com/capitalone/dashboard/service/CmdbServiceImpl.java:32]: performance is tested, added indexes in cmdb.

File: src/main/java/com/capitalone/dashboard/service/TestResultServiceImpl.java (hunk 1 above)
Comment (nameisaravind) [reply src/main/java/com/capitalone/dashboard/service/TestResultServiceImpl.java:607]: page data size is updated to the needed content.

**Review by rvema — APPROVED** [review]
lgtm

**Pull request merged**